etcdctl/ctlv3/command/auth_command.go

```diff
@@ -48,7 +49,21 @@ func authEnableCommandFunc(cmd *cobra.Command, args []string) {
 	}
 
 	ctx, cancel := commandCtx(cmd)
-	_, err := mustClientFromCmd(cmd).Auth.AuthEnable(ctx)
+	cli := mustClientFromCmd(cmd)
+	var err error
+	for err == nil {
+		if _, err = cli.AuthEnable(ctx); err == nil {
+			break
+		}
+		if err == rpctypes.ErrRootRoleNotExist {
+			if _, err = cli.RoleAdd(ctx, "root"); err != nil {
+				break
+			}
+			if _, err = cli.UserGrantRole(ctx, "root", "root"); err != nil {
```

Comment:
This doesn't create the root user, though. The idea is you can do:
```sh
etcdctl user add root
etcdctl auth enable
```

instead of
```sh
etcdctl user add root
etcdctl role add root
etcdctl user grant-role root root
etcdctl auth enable
```